vspace/src/main/java/edu/asu/diging/vspace/web/staff/SpaceOverviewController.java

```diff
@@ -0,0 +1,236 @@
+            }
+            List<T> tempSpaceToOtherlinkList = spaceToAllOtherlinksMap.get(spaceId);
+            tempSpaceToOtherlinkList.add(link);
+        });
+    }
+
+    /**
+     * creating json for space and module node in the spaceoverview graph
+     * 
+     * @param <T>
+     * @param request
+     * @param nodeList
+     * @param mapper
+     * @param nodeArray
+     * @param allNodeList
+     * @param isModule
+     */
+    private <T> void nodeFormation(HttpServletRequest request, List<T> nodeList, ObjectMapper mapper,
+            ArrayNode nodeArray, List<String> allNodeList, boolean isModule) {
+        nodeList.forEach(nodeval -> {
+            VSpaceElement nodeObj = isModule == true ? (Module) nodeval : (Space) nodeval;
+            ObjectNode node = mapper.createObjectNode();
+            node.put("name", nodeObj.getName());
+            StringBuilder linkPathBuilder = new StringBuilder();
+            linkPathBuilder.append(request.getContextPath());
+            String path = isModule == true ? ModuleController.STAFF_MODULE_PATH : SpaceController.STAFF_SPACE_PATH;
+            linkPathBuilder.append(path);
+            linkPathBuilder.append(nodeObj.getId());
+            node.put("link", linkPathBuilder.toString());
+            String imagePath = "";
+            if (!isModule) {
+                StringBuilder imagePathBuilder = new StringBuilder();
+                imagePathBuilder.append(request.getContextPath());
+                imagePathBuilder.append(ImageApiController.API_IMAGE_PATH);
+                imagePathBuilder.append(((Space) nodeObj).getImage().getId());
+                imagePath = imagePathBuilder.toString();
+            }
+            node.put("img", imagePath);
+            node.put("isModule", isModule);
+            if (!isModule) {
+                if (((Space) nodeObj).getSpaceStatus() != null
+                        && ((Space) nodeObj).getSpaceStatus().equals(SpaceStatus.UNPUBLISHED)) {
+                    node.put("isUnpublished", true);
+                } else {
+                    node.put("isUnpublished", false);
+                }
+                if (((Space) nodeObj).isHideIncomingLinks()) {
+                    node.put("isHideIncomingLinks", true);
+                } else {
+                    node.put("isHideIncomingLinks", false);
+                }
+            }
+            nodeArray.add(node);
+            allNodeList.add(nodeObj.getId());
+        });
+    }
+
+    /**
+     * creating json for source node to all target nodes in the spaceoverview graph
+     * 
+     * @param mapper
+     * @param linkArray
+     * @param allNodeList
+     * @param spaceLinkMap
+     */
+    private void sourceToTargetFormation(ObjectMapper mapper, ArrayNode linkArray, List<String> allNodeList,
+            Map<String, List<String>> spaceLinkMap) {
+        spaceLinkMap.forEach((k, v) -> {
+            if (!v.isEmpty()) {
+                v.forEach(e -> {
```

Comment:
use regular for loops; there is no gain in nesting `forEach`, it's just harder to read (especially with varialbe names `v` and `k`.